[lldb] Use Function::GetAddress in Module::FindFunctions

labath opened the pull request:
The original code resulted in a misfire in the symtab vs. debug info deduplication code, which caused us to return the same function twice when searching via a regex (for functions whose entry point is also not the lowest address).

labath committed:
[lldb] Use Function::GetAddress in Module::FindFunctions

The original code resulted in a misfire in the symtab vs. debug info
deduplication code, which caused us to return the same function twice
when searching via a regex (for functions whose entry point is also not
the lowest address).

jimingham commented:
This looks fine, but more generally, is there any good reason to ever use Function::GetAddressRange?  Shouldn't that just be removed and all the instances replaced with either GetAddress or get the ranges and iterate over them looking for whatever they wanted?

labath commented:
There isn't a good reason for using GetAddressRange, and removing it is what I'm trying to do. But rather than trying to do it all at once, I'm doing it in pieces as some of the changes (particularly those that need to iterate over a set of ranges) are not trivial. This way I can also verify that each change does what it's supposed to do.

I changed most of the trivial call sites when I introduced GetAddress (in #115836). In retrospect, this is probably trivial enough that it could have been a part of that, but I skipped it then because the mapping part looked scary.

jimingham reviewed (approved):
LGTM.  You might add a comment to GetAddressRange saying "I'm in the process of getting rid of this, don't use it in the meantime...".  But presumably you are also watching PR's to make sure nobody reintroduces its use.

labath commented:
I added a "DEPRECATED" comment, which isn't quite as strong, but sort of conveys the same thing. The function is called from a bunch of places, but I don't think there's that much churn around it for backsliding to be a problem.
